monai/inferers/splitter.py

```diff
@@ -58,142 +92,369 @@ def __call__(self, inputs: Any) -> Iterable[tuple[torch.Tensor, Sequence[int]]]:
+        pad_mode: str | None = "constant",
+        device: torch.device | str | None = None,
+        reader: str | BaseWSIReader | type[BaseWSIReader] | None = "OpenSlide",
+        **reader_kwargs: dict,
+    ) -> None:
+        if pad_mode and pad_mode != "constant":
+            raise ValueError(
+                f"The underlying wsi readers only support for constant padding. pad_mod={pad_mode} is given."
+            )
+
+        super().__init__(
+            patch_size=patch_size, overlap=overlap, offset=offset, filter_fn=filter_fn, device=device, pad_mode=pad_mode
+        )
+        # Set WSI reader
+        self._set_reader(reader, reader_kwargs)
+        if self.reader.backend.lower() not in ["openslide", "cucim"]:
+            warnings.warn(
+                f"WSIReader with {self.reader.backend.lower()} backend is not supported for efficiently loading patches. "
+                "This may cause an significant slow down and a large memory foot print. "
+                "Please use other backends such as 'OpenSlide' or 'cuCIM' instead."
+            )
+
+    def _set_reader(self, reader: str | BaseWSIReader | type[BaseWSIReader] | None, reader_kwargs: dict) -> None:
+        """
+        Set the WSI reader object based on the input reader
+
+        Args:
+            reader: the module to be used for loading whole slide imaging. If `reader` is
+
+                - a string, it defines the backend of `monai.data.WSIReader`. Defaults to cuCIM.
+                - a class (inherited from `BaseWSIReader`), it is initialized and set as wsi_reader.
+                - an instance of a class inherited from `BaseWSIReader`, it is set as the wsi_reader.
+        """
+        self.reader: WSIReader | BaseWSIReader
+        self.reader_kwargs = reader_kwargs
+        if isinstance(reader, str):
+            self.reader = WSIReader(backend=reader, **self.reader_kwargs)
+        elif isclass(reader) and issubclass(reader, BaseWSIReader):
+            self.reader = reader(**self.reader_kwargs)
+        elif isinstance(reader, BaseWSIReader):
+            self.reader = reader
+        else:
+            raise ValueError(f"Unsupported reader type: {reader}.")
+
+    def _get_patch(self, inputs: Any, location: tuple[int, ...], patch_size: tuple[int, ...]) -> Any:
+        patch, _ = self.reader.get_data(wsi=inputs, location=location, size=patch_size)  # type: ignore
+        return patch[None]
+
+    def get_input_shape(self, inputs: Any) -> tuple:
+        """
+        Return the input spatial shape.
+
+        Args:
+            inputs: either a tensor of shape BCHW[D], representing a batch of images,
+                or a filename (str) or list of filenames to the image(s).
+
+        Returns:
+            spatial_shape
+
+        """
+        wsi = self.reader.read(inputs)
+        level = self.reader_kwargs.get("level", 0)
+        return self.reader.get_size(wsi, level)
+
+    def get_padded_shape(self, inputs: Any) -> tuple:
```

Comment:
This method appears to be identical to the inherited one from `SlidingWindowSplitter`.

Reply:
Thanks for catching it.